packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -314,24 +313,34 @@ function startServerInNewWindow(
 
   if (process.platform === 'darwin') {
     try {
-      return execa.sync(
+      return await execa(
         'open',
         ['-a', terminal, launchPackagerScript],
         procConfig,
       );
     } catch (error) {
-      return execa.sync('open', [launchPackagerScript], procConfig);
+      return await execa('open', [launchPackagerScript], procConfig);
     }
   }
   if (process.platform === 'linux') {
+    // Awaiting this causes the CLI to hang indefinitely, so this must execute without .sync.
     try {
-      return execa.sync(terminal, ['-e', `sh ${launchPackagerScript}`], {
+      return await execa(terminal, ['-e', 'sh', launchPackagerScript], {
         ...procConfig,
         detached: true,
       });
-    } catch (error) {
-      // By default, the child shell process will be attached to the parent
-      return execa.sync('sh', [launchPackagerScript], procConfig);
+    } catch (_) {
```

Comment:
IIRC we can use optional catch binding:
```suggestion
    } catch {
```

Reply:
Oh cool, didn't know that one